batch/src/main/java/com/hartwig/batch/BatchOperation.java

```diff
@@ -0,0 +1,11 @@
+package com.hartwig.batch;
+
+import com.hartwig.batch.operations.CommandDescriptor;
+import com.hartwig.pipeline.execution.vm.VirtualMachineJobDefinition;
+import com.hartwig.pipeline.storage.RuntimeBucket;
+
+public interface BatchOperation {
+    VirtualMachineJobDefinition convert(String input, RuntimeBucket bucket, String instanceId);
```

Comment:
naming can be improved. The noun "Operation" and the verb "convert" don't really work together.